dot/parachain/availability-store/availability_store.go

```diff
@@ -4,61 +4,160 @@
 package availabilitystore
 
 import (
+	"bytes"
 	"context"
 	"encoding/binary"
 	"encoding/json"
 	"errors"
 	"fmt"
+	"time"
 
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
 	"github.com/ChainSafe/gossamer/internal/database"
 	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/erasure"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-availability-store"))
 
 const (
-	avaliableDataPrefix = "available"
+	availableDataPrefix = "available"
 	chunkPrefix         = "chunk"
 	metaPrefix          = "meta"
 	unfinalizedPrefix   = "unfinalized"
 	pruneByTimePrefix   = "prune_by_time"
+
+	tombstoneValue = " "
+
+	// Unavailable blocks are kept for 1 hour.
+	KEEP_UNAVAILABLE_FOR = time.Hour
+
+	// Finalized data is kept for 25 hours.
+	KEEP_FINALIZED_FOR = time.Hour * 25
+
+	// The pruning interval.
+	PRUNING_INTERVAL = time.Minute * 5
 )
 
+type BETimestamp uint64
```

Comment:
What's BE?

Reply:
big-endian, I've added a comment to the code.

Reply:
I kinda like to keep things generic, might help to extend it later. For example; calling it `Timestamp` and renaming the `ToBytes()` to `ToBEBytes()` would make it generic and we could add more apis on top of it easily!